### DISCUSSION TITLE
Prompt User before replacing army when switching Hero in Battle-Only

### DISCUSSION THREAD

**Discussion by pflugs30** [ihhub/fheroes2#4399] [Ideas]
I was playing FreeHeroes2 with my kids last night, and I was using the Battle-Only mode to given them practice scenarios.  I set up an army, primary and secondary skills, a spell book, and the artifacts for them.  Then, my daughter requested I change the Hero to a woman (understandable!).  I clicked on the Hero picture and selected a Sorceress.

Unfortunately, once the Hero was changed, all monsters, primary and secondary skills, spell book, and artifacts were reset to default.  We had to build everything all over again.  Note: I noticed that only the left (attacking) army was replaced.  The right (defending) army was unchanged.  However, when you change the right Hero, then everything on that side is replaced.

Instead of replacing the scenario when changing the Hero, I suggest we at least prompt the User to _choose_ whether they want to replace the scenario.  Or, better yet, leave it unchanged.

**Prior to changing Left Hero**
![Part 1 - Previous Army](https://user-images.githubusercontent.com/4409272/137004468-60a33f1e-4712-4c5a-9eb2-22d12dc986df.JPG)

**After changing Left Hero**
![Part 2 - New Army](https://user-images.githubusercontent.com/4409272/137004478-5b83c7cd-ed7e-4077-84bc-555481676f4d.JPG)

**Comment by LeHerosInconnu** [comment]
Hello @pflugs30,

There are many flaws in the "Battle Only" module at the moment and many improvements can be made to it.
I already have a long list for that. :)
But this module is not the priority at the moment, at least in my case. :)
